lib/server/db/seeds/connection.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Utility for adding configurable Connections seeds to the
+ * database.
+ *
+ * @module ConnectionSeed
+ */
+var User = require("../model/User");
+var Connection = require("../model/Connection");
+
+/**
+ * @typedef {Object} ConnectionSeedConfig
+ * @property {String} ConnectionSeedConfig.user - The from-User
+ * @property {String[]} ConnectionSeedConfig.friends - The to-Users
+ */
+
+/**
+ * Create connections between a user and a variable number of friends and
+ * returns a then-able Promise, passing the connections array to the next "then".
+ *
+ * @private
+ * @function createConnections
+ * @param {ConnectionSeedConfig} config - The name of the from-User
+ * @returns {Promise}
+ */
+function createConnections(config) {
+    return User
+        .findOne({
+            username: config.user
+        })
+        .then(function (user) {
+            return User
+            // find all the input users
+                .find({
+                    $or: config.friends.map(function (friend) {
+                        return {
+                            username: friend
+                        };
+                    })
+                })
+                // bundle up the user and friends to pass as data
+                .then(function (friends) {
+                    return {
+                        user: user,
+                        friends: friends
+                    };
+                });
+        })
+        .then(function (data) {
```

Comment:
For example, since this function needs the `user` object found above, we bundle the user and friends both in an object on `line 42`. That object is returned from that Promise and is piped to the next `then` in the pipeline (which is this one).